### PR TITLE
399 ug changes

### LINKED ISSUE #359: [PE-D][Tester B] Inaccurate command format for entity for add
It is stated in the application that "all" is a valid entity:

But in fact, it is not:

Should edit the command format to exclude "all".

Command to reproduce: "add" then "add all".

-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: gohsl99/ped#17

### LINKED ISSUE #399: UG changes
- specify role and skill under contact
- Add overview at the start of the features
- Add what constitute duplicate for contact
- Move defination of dul
- Update app UI image
- remove "Management"

### PR DESCRIPTION
**Pull request by KengHian**
- closes #399 
- closes #359

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -74,7 +80,7 @@ faster than traditional GUI apps.
 </div>
 
 ## Contact Management
-Contacts represent candidates applying for job positions. Each contact can have details such as name, phone, email, and relevant skills, aiding in recruitment processes.
+Contacts represent candidates applying for job positions. Each contact can have details such as name, phone, email, and relevant skills, aiding in recruitment processes. A contact's role represents the job that they are applying for, and is used when screening contacts with a certain job. A contact is uniquely identified by their phone number and email, therefore no two contacts can share a same phone number or email.
```
Comment (Quasant) [inline docs/UserGuide.md:83]: near the last line, no two contacts can share THE same phone number or email.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -39,11 +39,17 @@ faster than traditional GUI apps.
...
-6. Refer to the [Features](#features) below for details of each command.
+6. Refer to the sections below for details of each command.
 
 --------------------------------------------------------------------------------------------------------------------
 
-## Features
+## Overview
+TalentConnect offers management of multiple entities that work together to better help our users. To facilitate this, the commands in app can be categorized into two types, entity specific and non-entity specific. For entity specific commands, it is required to supply an entity after the action in the command. e.g. `add contact ...` command has an entity `contact` after the action `add`. 
+
+The valid entities in the system are `contact`, `job`, `company` and `all`. The first three of those refers to a specific entity, while `all` is used in certain commands to refer to all three entities. Do note that not all commands/actions works on all valid entities, kindly refer to [Command Summary](#command-summary) for the summary of what entities is supported in each action/command.
```
[the remark this answers is not in the document]
Comment (Quasant) [reply docs/UserGuide.md:49]: `Do note that not all commands/actons WORK on...`

`kindly refer to .... for the summary of what entities ARE supported...`

File: docs/UserGuide.md [hunk 3]
```diff
@@ -160,7 +163,7 @@ Examples:
 
 
 ## Company Management 
-Companies are organizations that provide job positions. Each company can have attributes such as address, billing date, and contact details, facilitating job creation and matching.
+Companies are organizations that provide job positions. Each company can have attributes such as address, billing date, and contact details, facilitating job creation and matching. A company is unique identify by its name (case-insensitive).
```
Comment (Quasant) [inline docs/UserGuide.md:166]: Last line, A company is UNIQUELY IDENTIFIED by...

**Review by Quasant — APPROVED** [review]
lgtm

File: docs/UserGuide.md (hunk 3 above)
Comment (Quasant) [reply docs/UserGuide.md:166]: by IT'S name

File: docs/UserGuide.md [hunk 4]
```diff
@@ -232,7 +233,7 @@ Examples:
 * If the company at index 1 has a job attributed to it, `delete company 1` will also delete the job.
 
 ## Job Management
-Jobs are positions offered by companies that contacts can apply for. Jobs contain details like monthly salary, description, and requirements, enabling precise candidate matching.
+Jobs are positions offered by companies that contacts can apply for. Jobs contain details like monthly salary, description, and requirements, enabling precise candidate matching. A job is unique identified by its name and company. Multiple openings for the same job position at a company can be represented using a single job, hence duplicates are not allowed.
```
Comment (Quasant) [inline docs/UserGuide.md:236]: uniquely identified by it's name ...

File: docs/UserGuide.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (KengHian) [reply docs/UserGuide.md:49]: Fixed!

File: docs/UserGuide.md (hunk 1 above)
Comment (KengHian) [reply docs/UserGuide.md:83]: Fixed!

File: docs/UserGuide.md (hunk 3 above)
Comment (KengHian) [reply docs/UserGuide.md:166]: Fixed the first comment.

**Pull request merged**